### ISSUE TITLE
vector of taxa for get_interactions?

### ISSUE THREAD

**Issue by jebyrnes** [ropensci/rglobi#43]
Hello! Having a great time using this for a research project. Related to my issue below about taxonomic synonyms, because I cannot search by a standard identifier, like an Aphia ID, I have to search my species and all of their synonyms. Currently, this is ~ 3K species. Whew!

However - get_interactions only takes one at a time. I have tried wrapping this in map, but, inevitably somewhere along the line I get a

```
 Error in open.connection(3L, "rb") : 
  Timeout was reached: [api.globalbioticinteractions.org:443] Connection timed out after 10000 milliseconds 
```

Currently I have put a Sys.sleep in between calls, and am trying to dial it up to the right length to avoid this. However, it occurs to me that submitting a vector of taxa rather than a single taxa at a time might alleviate this problem (although I do not know if it causes issues on the backend - but it should be lower cost than fielding 3K separate requests?)

I thought that maybe get_interaction_table() might be an end-run around this, but, even the example in the helpfile seems to not be working now, so I'm guessing there is a problem there.

```
> get_interaction_table(source.taxon.names = list("Aves"), target.taxon.names = list('Insecta'))
data frame with 0 columns and 0 rows
```

Still, for folk like me who want to build an interaction web from what is in globi instead of exploring an individual interaction, this might be helpful!

**Comment by jhpoelen** [comment]
Hey @jebyrnes thanks for taking the time to open this issue.

As a quick glance, I can see some separate issues:

1. a question about how to search by aphia id (worms) (see also #42 )
2. how to produce a list of interactions that involve on or more species from a name list with thousands of names on it. 
3. suspicious results produced by ```get_interaction_table(source.taxon.names = list("Aves"), target.taxon.names = list('Insecta'))```

Like #42 , I am hoping to respond to your separate questions/observations sooner rather than later. 

Meanwhile, I was wondering, would you be interested in become a co-author/ contributor to the rglobi package? It appears you have a great feel for what folks might be looking for in a package like rglobi.

**Comment by jebyrnes** [comment]
Yup! #2 is what I'm most interested in here, though!

**Comment by jhpoelen** [comment]
As noted on the GloBI data page at https://globalbioticinteractions.org/data - 

"[...] Exploratory, interactive queries can be executed through SPARQL and Cypher (see more examples) endpoints, GloBI Search/Browse pages, or by using the REST-y GloBI Web API. For those that use R, rglobi is available to explore interaction data. rglobi can also be used to execute Cypher queries.

For research or other data intensive project, please use GloBI’s stable versioned integrated data published via doi:10.5281/zenodo.3950589 or, perhaps even better, consider using the original underlying datasets. Please see the process page to better understand how GloBI integrates data so that you can make an informed decision on what data to use for your studies. [...]"

As you are probably aware web apis should be treated with great caution because they are, by design, highly dynamic, implement paging mechanisms to retrieve large amounts of data and, because of that, typically do not scale well and are expensive to use and maintain in the long run. 

In a recent workshop https://www.globalbioticinteractions.org/deadwood2021/ , I created some material that suggests some "big data" tools to help work with versioned/published data products instead of instable, yet playful, web apis. Perhaps the https://www.globalbioticinteractions.org/deadwood2021/13-day-two-part-two/ would help provide some insights in the bias / errors and scalability issues introduced by commonly used tools. 

Also, I've created a big data cheat sheet with some notes that may be of interest . https://www.globalbioticinteractions.org/deadwood2021/code/cheatsheet.txt . 

In other words - I'd suggest to only used published data products for research purposes, and leave the APIs for initial exploration and discovery.
 
Curious to hear your thoughts on this.

**Comment by jhpoelen** [comment]
Closing stale issue. @jebyrnes please feel free to re-open / comment if this is this a concern.
